### ISSUE TITLE
OutdatedDocumentation false positive for class primary constructor?

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#6362]
## Expected Behavior
No report. Maybe [it's just not clear what's missing?](https://github.com/detekt/detekt/issues/5561).

## Observed Behavior
```kotlin
/**
 * @param a A
 * @param b B
 */
|class C(
	val a: WebElement,
	val b: String,
)
```
at `|` character which is only there to signal the location:
> OutdatedDocumentation - [Documentation of C is outdated]

## Steps to Reproduce
 * I'm using `allRules = true` and have no mention of this rule in the config.
 * See code above.

## Context
I'm just confused :)

## Your Environment

* Version of detekt used: 1.23.0 / 1.23.1
* Version of Gradle used (if applicable): 8.3 RC1
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: Windows
* Link to your project (if it's a public repository): not necessary

**Comment by atulgpt** [comment]
Hi @TWiStErRob compliant code would be as below
```kotlin
/**
 * @property a A
 * @property b B
 */
class C(
    val a: WebElement,
    val b: String,
)
```

since you have added `val` not those `a` and `b` are property of the class, not the param.

Although, I agree that outdated rule clearly mentions what is wrong in the documentation. Can you add these details in https://github.com/detekt/detekt/issues/5561 and should we close this issue?

**Comment by TWiStErRob** [comment]
Indeed `@property` works, but there's something fishy, because [private properties shouldn't be documented](https://detekt.dev/docs/1.23.0/rules/comments/#commentoverprivateproperty), right? (
```kotlin
/**
 * @property a A
 * @param b B
 */
class C(
    val a: WebElement,
    private val b: String,
)
```

detekt only accepts:
```kotlin
/**
 * @property a A
 * @property b B
 */
class C(
    val a: WebElement,
    private val b: String,
)
```

I played around a bit in IDEA "View > Quick Documentation" and they treat both param/property interchangeably. The only difference I saw was how the constructor labels its parameters according to `@` we use.

![image](https://github.com/detekt/detekt/assets/2906988/38a538ed-87bc-482b-a24a-85359de3c9f2)

Even though it's a constructor call, it shows the property docs.

---

I think the rule should be changed to allow primary constructors to be documented by `@param`, after all it's not outdated at this point, they're just documenting the constructor and not the properties.

**Comment by atulgpt** [comment]
Hi, @TWiStErRob did you also try `allowParamOnConstructorProperties`?

**Comment by TWiStErRob** [comment]
rtfm 🤦‍♂️

**Comment by atulgpt** [comment]
Hi @TWiStErRob, although outdated documents should prevent report on below code. Right?
```
/**
 * Doc
 * @property a desc
 */
class A(
    private val a: String,
)
```

Currently, there is not warning associated with it.
